Source/WebKit/UIProcess/Launcher/haiku/ProcessLauncherHaiku.cpp

```diff
@@ -26,142 +26,76 @@
-// of creating yet another BHandler there.
-class ProcessLauncherHandler: public BHandler
-{
-    public:
-        ProcessLauncherHandler(ProcessLauncher* launcher)
-            : BHandler("process launcher")
-            , m_launcher(launcher)
-        {
-        }
-
-        void MessageReceived(BMessage* message)
-        {
-            switch(message->what)
-            {
-                case 'inig':
-                    GlobalMessage(message);
-                    break;
-                default:
-                    BHandler::MessageReceived(message);
-                    break;
-            }
-        }
-
-    private:
-
-        void GlobalMessage(BMessage* message)
-        {
-            WTF::ProcessID processID = message->FindInt64("processID");
-            BMessenger messenger;
-            message->FindMessenger("messenger", &messenger);
-            IPC::Connection::Identifier connectionIdentifier(std::move(messenger));
-            connectionIdentifier.m_isCreatedFromMessage = true;
-            m_launcher->didFinishLaunchingProcess(processID, connectionIdentifier);
-
-            // Our job is done!
-            // FIXME or is it? maybe keep this around until the processLauncher is destroyed?
-            delete this;
-        }
-
-        ProcessLauncher* m_launcher;
-};
-
-static BHandler* GetProcessLauncherHandler(ProcessLauncher* launcher)
-{
-    BHandler* handle = nullptr;
-    handle = new ProcessLauncherHandler(launcher);
-    BLooper* looper = BLooper::LooperForThread(find_thread(NULL));
-    looper->AddHandler(handle);
-
-    return handle;
-}
-
-static status_t GetRefForPath(BString path, entry_ref* pathRef)
-{
-    BEntry pathEntry(path);
-    if(!pathEntry.Exists())
-        return B_BAD_VALUE;
-
-    status_t result = pathEntry.GetRef(pathRef);
-    if(result != B_OK)
-        return result;
-
-    return B_OK;
-}
-
 void ProcessLauncher::launchProcess()
 {
-    BString executablePath;
+    IPC::SocketPair socketPair = IPC::createPlatformConnection(
+        IPC::PlatformConnectionOptions::SetCloexecOnClient | IPC::PlatformConnectionOptions::SetCloexecOnServer);
 
+    char* executablePath;
     switch (m_launchOptions.processType) {
     case ProcessLauncher::ProcessType::Web:
-        executablePath = executablePathOfWebProcess();
+        executablePath = strdup(executablePathOfWebProcess().ascii().data());
         break;
     case ProcessLauncher::ProcessType::Network:
-        executablePath = executablePathOfNetworkProcess();
+        executablePath = strdup(executablePathOfNetworkProcess().ascii().data());
         break;
     default:
         ASSERT_NOT_REACHED();
         return;
     }
 
-    BString processIdentifierString;
-    processIdentifierString.SetToFormat("%" PRIu64, m_launchOptions.processIdentifier.toUInt64());
+    char processIdentifierString[16];
+    snprintf(processIdentifierString, sizeof processIdentifierString, "%" PRIu64, m_launchOptions.processIdentifier.toUInt64());
```

Comment:
Here as well, why not use BString and let it handle the memory allications for us?